nixos/modules/services/networking/consul.nix

```
@@ -199,18 +199,18 @@ in
             (filterAttrs (n: _: hasPrefix "consul.d/" n) config.environment.etc);
 
         serviceConfig = {
-          ExecStart = "@${cfg.package}/bin/consul consul agent -config-dir /etc/consul.d"
+          ExecStart = "@${lib.getExe cfg.package} consul agent -config-dir /etc/consul.d"
             + concatMapStrings (n: " -config-file ${n}") configFiles;
           ExecReload = "${pkgs.coreutils}/bin/kill -HUP $MAINPID";
           PermissionsStartOnly = true;
           User = if cfg.dropPrivileges then "consul" else null;
           Restart = "on-failure";
           TimeoutStartSec = "infinity";
         } // (optionalAttrs (cfg.leaveOnStop) {
-          ExecStop = "${cfg.package}/bin/consul leave";
+          ExecStop = "${lib.getExe cfg.package} leave";
         });
 
-        path = with pkgs; [ iproute2 gnugrep gawk consul ];
+        path = [ pkgs.iproute2 pkgs.gnugrep pkgs.gawk cfg.package ];
```

```suggestion
        path = with pkgs; [ iproute2 gnugrep gawk cfg.package ];
```
we don't need to repeat that. Also isn't grep already in PATH by default?

---

`grep` is in the path by default, yes.
`gnugrep` seems to be listed in consul's service definition since it's addition to nixpkgs in #4280 (2014).
Removed the reference as requested.

As for why I chose to drop the `with pkgs;`:
I saw that in another PR review to make it more clear where the var is coming from, a few weeks ago.

Undid that as well, as requested.